src/dotnet/commands/dotnet-add-p2p/Program.cs

```diff
@@ -0,0 +1,174 @@
+            });
+
+            try
+            {
+                return app.Execute(args);
+            }
+            catch (GracefulException e)
+            {
+                Reporter.Error.WriteLine(e.Message.Red());
+                app.ShowHelp();
+                return 1;
+            }
+        }
+
+        // There is ProjectRootElement.TryOpen but it does not work as expected
+        // I.e. it returns null for some valid projects
+        public static ProjectRootElement TryOpenProject(string filename)
+        {
+            try
+            {
+                return ProjectRootElement.Open(filename, new ProjectCollection(), preserveFormatting: true);
+            }
+            catch (Microsoft.Build.Exceptions.InvalidProjectFileException)
+            {
+                return null;
+            }
+        }
+
+        public static ProjectRootElement GetProjectFromFileOrThrow(string filename)
+        {
+            if (!File.Exists(filename))
+            {
+                throw new GracefulException($"Provided project `{filename}` does not exist.");
+            }
+
+            var project = TryOpenProject(filename);
+            if (project == null)
+            {
+                throw new GracefulException($"Invalid project `{filename}`.");
+            }
+
+            return project;
+        }
+
+        // TODO: improve errors
+        public static ProjectRootElement GetProjectFromDirectoryOrThrow(string directory)
+        {
+            DirectoryInfo dir;
+            try
+            {
+                dir = new DirectoryInfo(directory);
+            }
+            catch (ArgumentException)
+            {
+                throw new GracefulException($"Could not find project or directory `{directory}`.");
+            }
+
+            if (!dir.Exists)
+            {
+                throw new GracefulException($"Could not find project or directory `{directory}`.");
+            }
+
+            FileInfo[] files = dir.GetFiles("*proj");
+            if (files.Length == 0)
+            {
+                throw new GracefulException($"Could not find any project in `{directory}`.");
+            }
+
+            if (files.Length > 1)
+            {
+                throw new GracefulException("Found more than one MSBuild project in the current directory. Please specify which one to use.");
+            }
+
+            FileInfo projectFile = files.First();
+
+            if (!projectFile.Exists)
+            {
+                throw new GracefulException($"Could not find any project in `{directory}`.");
+            }
+
+            var ret = TryOpenProject(projectFile.FullName);
+            if (ret == null)
+            {
+                throw new GracefulException($"Found a project `{projectFile.FullName}` but it is invalid.");
+            }
+
+            return ret;
+        }
+
+        public static int AddProjectToProjectReference(ProjectRootElement root, string framework, IEnumerable<string> refs)
```

Comment:
How about instead of refs, we call it referencesToAdd? Very explicit. Remember to keep the line at 120 characters.